### PR TITLE
Refactor DoIfCostPaid, inform on not paying the cost.

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Devouring Sugarmaw]]

File: Mage/src/main/java/mage/abilities/effects/common/DoIfCostPaid.java [hunk 1]
```diff
@@ -100,69 +100,64 @@ public boolean apply(Game game, Ability source) {
...
-            } else if (!otherwiseEffects.isEmpty()) {
-                for (Effect effect : otherwiseEffects) {
-                    effect.setTargetPointer(this.targetPointer);
-                    if (effect instanceof OneShotEffect) {
-                        result &= effect.apply(game, source);
-                    } else {
-                        game.addEffect((ContinuousEffect) effect, source);
-                    }
+            } else {
+                if (canPay) {
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/effects/common/DoIfCostPaid.java:118]: I think this leaks unnecessary info? Some costs use hidden information e.g. reveal a card from your hand. So better to use the "did not" inform text in all cases.

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage/src/main/java/mage/abilities/effects/common/DoIfCostPaid.java (hunk 1 above)
Comment (Susucre) [reply Mage/src/main/java/mage/abilities/effects/common/DoIfCostPaid.java:118]: Oh right. Forgot about costs from hidden zones.

**Comment by theelk801** [comment]
are the `DoIfCostPaid` changes necessary to this card? if not, would you mind breaking them out to a different PR?

**Comment by Susucre** [comment]
#10969 for the simple implementation of the card. I'll repurpose the current PR to the DoIfCostPaid cleanup.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**